### ISSUE TITLE
LGTM.com - false positive

### ISSUE THREAD

**Issue by pepoluan** [github/codeql#4918]
**Description of the false positive**

LGTM.com alerted about "illegal raise", with the description "Illegal class 'NoneType' raised; will result in a TypeError being raised instead."

Refer to the [screenshot on Twitter](https://pbs.twimg.com/media/Eq8c3vTUwAEqlIU?format=png).

Code replicated:

```python
        if self._thread_exception is not None:  # pragma: nowsl
            # See comment about WSL1.0 in the _run() method
            raise self._thread_exception
```

As you can see, it is simply _impossible_ for `self._thread_exception` to be `None` when it's raised; the `if` statement on the first line guarantees that.

This occurred **twice** in the same file; see URLs below:

**URL to the alert on the project page on LGTM.com**

https://lgtm.com/projects/g/aio-libs/aiosmtpd/snapshot/1dc00d77a8767369d079567e4240b757a3e96d7b/files/aiosmtpd/controller.py?sort=name&dir=ASC&mode=heatmap#x5de8abb8d304cf9:1

and

https://lgtm.com/projects/g/aio-libs/aiosmtpd/snapshot/1dc00d77a8767369d079567e4240b757a3e96d7b/files/aiosmtpd/controller.py#x95bc8f7524427423:1

**Comment by yoff** [comment]
I agree that those are (somewhat hilarious) false positives. I note that the alerts are old and [not currently present](https://lgtm.com/projects/g/aio-libs/aiosmtpd/?mode=list) (congratulations on zero alerts, by the way!). That would suggest that the issue is not with that particular alert but rather due to some instability in a component of our analysis. Fortunately, we are already actively working on the most likely candidate. Unfortunately, this may take some time, as it is foundational work.

**Comment by pepoluan** [comment]
> I note that the alerts are old and [not currently present](https://lgtm.com/projects/g/aio-libs/aiosmtpd/?mode=list) (congratulations on zero alerts, by the way!)

Yup, I quashed all alerts on `master` ... this appeared when I last pushed to a PR here: https://github.com/aio-libs/aiosmtpd/pull/202

Fortunately, that PR is still a work in progress so there's no urgency in merging.

I'd be sad if the zero alert record of aiosmtpd is broken, though...
